File: 3.c
/*Project Euler solution 3, author: http://github.com/TerkaSlan */
#include <stdio.h>
#include <time.h>

int main()
{
    clock_t start, end;
    float time_passed;
    start = clock();
    unsigned long long i, num = 600851475143ULL;

    for(i=2ULL; i<num; i++){
        // Found the divisor, divide num until not divisible anymore
        while(num % i == 0){
            num /= i;
        }
	
    }

    printf("Found: %llu", num);

    end = clock();
    time_passed = (end-start)*1.0/CLOCKS_PER_SEC;
    printf("\nThe execution took me %f seconds\n", time_passed); // 0.000317s
    return 0;
}

File: 3.py
import math
#Project Euler solution 3, author: http://github.com/TerkaSlan #
from time import clock

def is_prime(n):
    if n < 0:
        return False
    for j in range(2, int(math.sqrt(n)+1),1):
        if n%j == 0:
            return False
    return True


limit = 600851475143
i = 1
start = clock()
while i*i <= limit:
    i += 2 #ignoring all even numbers
    if limit % i == 0:
        if is_prime(i):
            max = i
end = clock()
print 'Found: ', max
print 'It took me ', end-start, ' seconds' # 0.096s
